sdk/iot/hub/samples/CMakeLists.txt

```
@@ -17,78 +17,95 @@ if(NOT OpenSSL_FOUND)
   find_package(OpenSSL REQUIRED)
 endif()
 
-set (sample_link_libraries 
-  az_iot_hub
-  az_iot_common
-  az_core
-  ${PAL}
-  eclipse-paho-mqtt-c::paho-mqtt3cs-static
-  OpenSSL::SSL
-  OpenSSL::Crypto
-)
-
 # C2D Sample
-add_executable (
-  paho_iot_hub_c2d_example 
+add_executable (paho_iot_hub_c2d_example
   ${CMAKE_CURRENT_LIST_DIR}/src/paho_iot_hub_c2d_example.c
-  )
+)
+
+target_link_libraries(paho_iot_hub_c2d_example
+  PRIVATE
+    az::core
+    az::iot::hub
+)
+
 
 target_link_libraries(
   paho_iot_hub_c2d_example PRIVATE
```

consistency nit: move private to newline, like elsewhere and match the formatting.

move the target name up to line 36.

```suggestion
target_link_libraries(paho_iot_hub_c2d_example
  PRIVATE
```